.pipelines/ci/templates/build-powertoys-steps.yml

```diff
@@ -83,6 +83,24 @@ steps:
 - task: VisualStudioTestPlatformInstaller@1
   displayName: Ensure VSTest Platform
 
+- task: Cache@2
```

Comment:
Should these be guarded behind the cache parameter? Or is this harmless because it isn't caching build outputs but static verifiable files that can be guaranteed to be the same across all runs?

Reply:
I intentionally sent this PR from my fork to test this :).

The `Cache@2` task is special and is able to elevate its own permissions on-the-fly it seems. Presumably this is safe since it's not running arbitrary user code. There is also cache isolation and scoping in place so forks/PR/other branches can't mess with each other and there's a hierarchy and security around it ([docs](https://learn.microsoft.com/en-us/azure/devops/pipelines/release/caching?view=azure-devops#cache-isolation-and-security))

Reply:
The reason project caching is different is because although we're using the same backend (pipeline caching), we're just passing the access token as an env var to the MSBuild call and our MSBuild plugin is using that. From the pipeline's perspective, we're just kinda using the access token willy nilly so it doesn't have permissions to do much, including talk to pipeline caching.

Ideally one day the access token could be given read-only access to the pipeline cache, even from forks, but that would require some server-side changes in AzDO.